app/data_model/answer_store.py

```diff
@@ -195,27 +196,37 @@ def map(self, group_id=None, block_id=None, answer_id=None, group_instance=None,
         :return:
         """
         result = {}
-        use_filter = (group_id or block_id or answer_id or group_instance or answer_instance) is not None
-        answers = self.filter(group_id, block_id, answer_id, group_instance, answer_instance) if use_filter else self.answers
-
-        for answer in answers:
+        for answer in self.filter(group_id, block_id, answer_id, group_instance, answer_instance):
             answer_id = answer['answer_id']
             answer_id += "_" + str(answer['answer_instance']) if answer['answer_instance'] > 0 else ''
 
             result[answer_id] = answer['value']
 
         return OrderedDict(sorted(result.items(), key=lambda t: natural_order(t[0])))
 
-    def remove(self, answer):
+    def remove_answer(self, answer):
```

Comment:
Yeah, @iwootten has just mentioned it as well. We need this merged as it's holding up other pull requests, so I'll create a card for the potential refactoring.